### DISCUSSION TITLE
PSRule testing of parameter files contains to many parameters

### DISCUSSION THREAD

**Discussion by JonasCordsen** [Azure/PSRule.Rules.Azure#1990] [Ideas]
Hello @BernieWhite 

We are running PSRule and found out that if our deployment parameter file has a parameters that the bicep file "main.bicep" does not cause an error.

This does give us an error when the bicep files then deploys to Azure.
Is this by design, or is there an option that we can use, in order to get this to be testet?

**Comment by BernieWhite** [comment]
@JonasCordsen I'm not sure if I correctly understand. If I'm understanding correctly:

- You have a parameter file that has additional parameters that are not defined within the `main.bicep`.

I'm unsure if you are asking if:

- PSRule should fail but Azure does.
- PSRule fails but Azure does not.

---

The goal is for PSRule to behave similar to Azure, so if Azure errors then we should add a rule to detect this however I doesn't a rule currently exists for this.

**Reply by JonasCordsen** [reply]
@BernieWhite 

You understand correctly.

And this is What i am looking for

PSRule should fail but Azure does.

Would be a great thing to have

**Reply by BernieWhite** [reply]
Created a rule request for this issue under #1993.
